packages/container/lib/container.js

```diff
@@ -237,7 +237,21 @@ function injectionsFor(container, fullName) {
                                    registry.getTypeInjections(type),
                                    registry.getInjections(fullName));
   injections._debugContainerKey = fullName;
-  injections.container = container;
+
+  if (isEnabled('ember-registry-container-reform')) {
+    Object.defineProperty(injections, 'container', {
```

Comment:
Is this ran once for each object looked up?  Does that pose a performance issue? I am unsure, I'd love others' thoughts...